### ISSUE TITLE
Tags on upload

### ISSUE THREAD

**Issue by legionfrontier** [rr-/szurubooru#188]
Greetings, following the https://github.com/rr-/szurubooru/issues/182, can we have another options that doesnt involves the usage of a hilariously overcomplicated program to tag stuff and mass upload images to the booru? because sure isnt fun to do so on 50k images manually when I already have most tags on the filename anyway, I will greatly appreciate it

**Comment by legionfrontier** [comment]
@rr- https://github.com/mserajnik/szuruporter @mserajnik 's tool makes uses of Hydrus that is hilariously overcomplicated imo, the images are already like this: "blue_hair girl smiling hugging rem blushing.jpg" so I wish to use the filename already to get tags on upload

**Comment by mserajnik** [comment]
@legionfrontier 

So what you want to have is an option in szurubooru itself (like a checkbox in the upload view) that let's you automatically tag images based on their filenames, right?

So, uploading a file called `blue_hair girl smiling hugging rem blushing.jpg` would create a post that has the following tags:

```
blue_hair
blushing
girl
hugging
rem
smiling
```

@rr- This is actually quite different from #182 where the request was to have a textarea to enter tags. I don't think it's such a bad idea, there might be more people who have descriptive tags as their filenames who could appreciate such a feature when migrating. What's your opinion on it?

**Comment by noirscape** [comment]
FWIW, myimouto (which was my go to booru before szurubooru got docker support), supported this feature in a way, where you could upload files to a folder and it would upload them if you went to the batch import page, with tags coming from both the filename and folder structure.

I say in a way, because I never got this working for the folder structure (so much for that feature), but did get it working for the folder.

Having this as an option for the upload form would be very useful to me (over implementing this batch feature remotely/as a script) as it would save me the time of having to use `scp` to upload these files to the server first and then run the batch import script on them

**Comment by legionfrontier** [comment]
@mserajnik yes mate, exactly I want that

@rr- yeah so please reopen my issue plox, seems we have 2 folk over that like the idea too

**Comment by rr-** [comment]
>@rr- This is actually quite different from #182 where the request was to have a textarea to enter tags. I don't think it's such a bad idea, there might be more people who have descriptive tags as their filenames who could appreciate such a feature when migrating. What's your opinion on it?

No. It's a feature creep.

Just use the API and upload the posts however you want…

**Comment by mserajnik** [comment]
@rr- 

> No. It's a feature creep. (I can see a magnitude of issues such as "hey all my images are tagged `konachan.com`" "hey my tags contain weird characters")

Fair enough, I didn't think about the potential issues. Going further, you could only combat such problems by having a blacklist of words/phrases to exclude when parsing the tags from the filename. And having such a thing (with it's own view, savable settings and everything) would then really bloat up szurubooru just for such a niche feature.

@legionfrontier 

Since @rr- has dismissed adding it as an option directly in szurubooru, let me know if you want to use my tool instead and if I should add that option (parsing tags from the filename when importing). Best to just [open a new issue](https://github.com/mserajnik/szuruporter/issues/new) there for that feature request.

Otherwise, as @rr- has said, you can always use the API to upload stuff however you want and putting together a simple upload script (like mine) really doesn't take much with all the HTTP client libraries available these days for any language.
